x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/action/TransportExplainLifecycleAction.java

```diff
@@ -98,7 +98,7 @@ protected void doMasterOperation(ExplainLifecycleRequest request, String[] concr
                 }
             }
             final IndexLifecycleExplainResponse indexResponse;
-            if (Strings.hasLength(policyName)) {
+            if (Strings.hasLength(policyName) && Strings.hasLength(currentPhase)) {
```

Comment:
this was caught because there is a time window where the index has a policy name, but is not really 
managed by ILM just yet since it doesn't have its step info set. I thought it would be sufficient to keep this 
change small and add this extra check to verify that the lifecycle state exists. One way to check that it exists is to verify that there is at least one field set, the current phase.